### ISSUE TITLE
Not getting Streetpasses?

### ISSUE THREAD

**Issue by github-hudsucker-com** [h0bd0b1in/XZone#1]
I have a MacBook Pro with a wired connection to my router and running today's version for XZone with wifi for the New 3DS. My 3DS has Internet connectivity through this connection, but it's not receiving any Streetpass notices. I have the interval set to every 5 minutes (tried 2 as well), but still, nothing. The Streetpass notifier on the 3DS comes on when I flip the lid, but it never cycles through like I've seen in the past.

Any ideas?

**Comment by ixalsera** [comment]
A few questions ...
- Is XZone running with output in terminal so it shows that it is actually cycling through the MACs? Could you post the output in to something like PasteBin for me to have a look at, or pasting it in here? (Logging is coming in a future build)
- What speed is your DSL/Broadband connection? Although this shouldn't be an issue, sometimes if your line is throttled or slow, it takes longer to sync with the NZone servers.
- Which SSID were/are you using? 

If you just updated to the latest version, and the cycle started from :00 again (because of no .plist from the previous versions), it's possible that the MACs are on cooldown on your 3DS. I'm not sure about the n3DS, but I know the 3DS has a MAC cooldown of around 8 hours (so you won't get a torrential influx of StreetPasses from your friends constantly). Cycling at 2 minutes totals 8.5 hours for all the MACs (not including Nintendo World MACs) which avoids the cooldown issue, but if the script was started without the .plist (or the .plist was reset) or without a specific starting MAC octet, it starts from the beginning. If 8 hours haven't passed since it last started, your 3DS will ignore those MACs.

**Comment by github-hudsucker-com** [comment]
Yes, xzone is running with output in the terminal. I've pasted a sample of it below.

My cable connection is 50Mb/12Mb. I can routinely max it out to test it, but during testing of xzone, I made sure nothing was really utilizing my bandwidth.

I've tried a bunch of the SSIDs in the app as well as some custom ones. By default, I've been trying to use the attwifi SSID.

Also, I thought it might be an issue with my New 3DS, so I took it with me to a local Gamestop and within minutes of being there, I picked up six new StreetPass tags. I also live near a middle school so I can usually get a tag or two from kids walking home from school and passing by my apartment.

Any other ideas? I'm not trying to blame you or the script for this. I'm just frustrated that I can't get it to work. I'm sure it's something on my end, but I'm just running out of things that I can think of that might be causing the problem.

**_s-MBP:XZone-master ***_***$ ./xzone -r
We neet root privileges for this!
Password:

Would you like to include the Nintendo World MACs in the cycle? Y or N
y

Including Nintendo World MACs.

Please enter the time (in minutes) you wish the script
to cycle through addresses:  30
Script will cycle addresses every 30 minutes (1800 seconds).

If you would like to start at a specific MAC, enter the final
2-digit hex value (octet) here. (This is useful if you have terminated
the script and wish to pick up where you left off) Otherwise, just hit
return.
40

Starting at 4e:53:50:4f:4f:40

Which SSID would you like to use?
1. attwifi      2. NZ@McD1      3. (More ...)
1
SSID set to attwifi
Script Starting!

Spoofing MAC!
MAC spoofed to 4e:53:50:4f:4f:40
Resetting Wi-Fi ...
Wi-Fi reset!
Spoofing MAC!
MAC spoofed to 4e:53:50:4f:4f:41
Resetting Wi-Fi ...
Wi-Fi reset!
Spoofing MAC!
MAC spoofed to 4e:53:50:4f:4f:42
Resetting Wi-Fi ...
Wi-Fi reset!
Spoofing MAC!
MAC spoofed to 4e:53:50:4f:4f:43
Resetting Wi-Fi ...
Wi-Fi reset!

<...REPEAT...>

**Comment by ixalsera** [comment]
- Have you recently updated OS X? By which I mean a major upgrade (10.7 to 10.8, for example).
- Have you tried restarting Internet Sharing by toggling it off and on again in System Preferences?
  Is your machine definitely broadcasting on the SSID you select? (Ignoring the output to Terminal, which is actually pretty inefficient in the current builds of XZone as it will tell you it's set the SSID even if setting it failed, or it set the wrong file. I'll fix this in a future build.)
- Have you tried rebooting your machine?
- Another thing I've just thought, when you open the 3DS, does it automatically connect to the internet (i.e. does the WiFi notifier turn blue in the top left corner)?
- Is Internet Sharing set to have no security in System Preferences. No idea why I haven't built this in to the code yet, obviously slipped my mind. But double check that security is set to 'None'.

There's probably something silly I'm overlooking (I spent ages trying to work out why the hell the cleanup code was being called twice before realising it was just a matter of where I'd positioned the `trap`) so I'm just taking you through the basic troubleshooting steps, although I'm sure you've done them already.

**Comment by github-hudsucker-com** [comment]
OSX is running at 10.10.2. I updated it before even trying to run XZone.
I have toggled Internet Sharing off and on several times. It is broadcasting on the proper SSID, since before I run the script, I can see the SSID on my phone that it's broadcasting the old name, then when the script starts, it shows the attwifi SSID.
I've rebooted the laptop several times.
When I open the N3DS, it does show that it's connected to the wifi (I've removed all three wifi connections configured and set up the new one as the only network to connect to). It also briefly shows the StreetPass green icon for a few seconds when the lid is first opened, but then never shows it again, but I understand that's the new normal process.
There is no security set on the wifi connection being shared from the laptop.

I'm sure there's something stupid that I'm overlooking too, but I just can't put my finger on it.

**Comment by ixalsera** [comment]
All that info does help narrow it down a bit. I'm going to open up a beta track tomorrow and push a new build to that I'd like you to try. I won't be able to test much myself this coming weekend but I'll do what I can.

**Comment by github-hudsucker-com** [comment]
That sounds awesome. I'll keep an eye out for it and do some testing and let you know what I see. Thanks for all your hard work.

**Comment by ixalsera** [comment]
Too much happened over the weekend.

There's a beta track now open with v0.5.3b up. I've tweaked a couple of things in it, so give it a whirl and let me know if you come right.

<strong>Edit:</strong> Just noticed you said you set up the new network as the only one to connect to. You shouldn't have any network connections set up on your n3DS. The 3DS prioritises saved connections over StreetPass connections (which makes sense, since normally you would be at an event or away from a place where you usually have WiFi) so you won't receive any StreetPass data. Make sure you remove ALL your connections from your n3DS and then try.

**Comment by github-hudsucker-com** [comment]
The whole wifi comment makes perfect sense. That was probably the boneheaded mistake I was making to begin with.

I've removed all wifi connections and am running the 0.5.3b with all the default settings. I'll let it run for a few hours and see what happens. I will say that I do like the verbose output of the previous version so you can see when the MAC address changes. You've probably got it as an option, but I haven't had time to look through the code yet.

Thanks again! I'll post again when it's done running for a while.

**Comment by github-hudsucker-com** [comment]
Been about 8 hours now and still no StreetPass tags with the new beta script.

I just got a new Raspberry Pi B, so I'm going to give SpillPass-Pi a try and see if that works so I can rule out if it's my N3DS, MBPro or just me being an idiot.

**Comment by ashgoldprint** [comment]
Hi just wanted to say that I love your script! Works better than any other out there for Mac. 

Also I went and looked at your web comic via searching your username - it's so unique and adorable! You're clearly a talented person and I hope you keep creating things using those talents :) I don't even know if you'll get this email lmao I could not for the life of me find a way to contact you through GitHub

**Comment by ixalsera** [comment]
@github-hudsucker-com, did you manage to get SpillPass working or is XZone working for you again? You can run XZone as verbose by launching it with the `-v` option. There's a new beta build out that fixes a few minor bugs but nothing that would affect receiving StreetPass data I don't think (unless you had corrupt preferences).

As an extra check, I stopped receiving StreetPasses for a while too. Just to make sure it wasn't XZone, I stopped XZone, opened up System Settings on my 3DS, navigated to "Data Management", "Nintendo 3DS", "Extra Data" and deleted the "Nintendo Zone Viewer" data block. Then, after switching my 3DS off and on again I restarted XZone using "NZ@McD1" as the SSID, opened "Nintendo Zone Viewer" and let it write the extra data again. That got me back up and running.

@ashgoldprint, thanks for the support! :) This is my first public project so I'm glad you like it. And if by web comic you mean Nemi, she isn't actually my creation. I wish I was that talented. :( She's a Norwegian comic strip created by Lise Myhre. I just hosted some English translations on my old blog.

**Comment by github-hudsucker-com** [comment]
No, still no luck. The SpillPass solution isn't working because it won't boot and all the troubleshooting options they give don't help (although other Raspberry OS's are booting fine).

As for XZone, I went through the exact steps above with your newest beta and it still won't pull anything down. I know the N3DS is working fine since if I take it out with me in public or to a game store, I pick up lots of Streetpasses, but if I let it sit here at home, the green Streetpass logo on the top screen flashes when I open the case, but it never picks up any Streetpasses.

My next step is to try the Windows virtual router solution. I'm spending a lot of time trying to figure out something that should be pretty trivial. But thank you for being so communicative.

**Comment by ixalsera** [comment]
The green "StreetPass" should only flash briefly and then switch to the regular blue "Internet". If it isn't switching to Internet then there is no internet route between your wired internet connection and the Wi-Fi, which would be the cause of the lack of StreetPasses. Do you have a very strict firewall set up on your network? Is the Firewall enabled in OS X?

A couple more things to check, open System Preferences and the Network pane. Ethernet should show as connected (the settings here shouldn't need to be changed so long as it shows Ethernet as connected with a green light). Wi-Fi should be orange with a Self Assigned IP and if you click it, should show "Internet Sharing: On" under the "Network Name" dropdown. If you click the "Advanced..." button in the lower right, and switch to the TCP/IP tab, make sure "Configure IPv4" is set to "Using DHCP". If it isn't, the Subnet Mask could be different and we need it to be "255.255.0.0" for it to correctly route information between the network that Internet Sharing creates (192.168.2.1/255) and your home network.

Does your Wi-Fi menu bar item show that sharing is enabled? Can you pick up the "attwifi" or "NZ@McD1" network on any other devices and browse the web?

With XZone running, can you open a separate terminal window and type `ifconfig >> ~/Desktop/ifconfig.log` and then paste the log it dumps on your desktop here? I just want to check that the interface bindings are correct.

**Comment by github-hudsucker-com** [comment]
No firewall running (just a Netgear router running DD-WRT) and I even disabled SPI on the router but that didn't help. 

The StreetPass logo flashes green when the lid is opened on the N3DS and then switches to blue Internet. I'm able to browse the Internet on an iPhone 6, an iPad Air 1 and the N3DS, but still can't access the Nintendo Zone. I get an error code 003-1099.

All the network settings you mentioned are set accordingly (I can post screenshots if you want) and the Sharing settings are set up properly (again, screenshots if you want).

Here are the results of the ifconfig....

lo0: flags=8049<UP,LOOPBACK,RUNNING,MULTICAST> mtu 16384
    options=3<RXCSUM,TXCSUM>
    inet6 ::1 prefixlen 128
    inet 127.0.0.1 netmask 0xff000000
    inet6 fe80::1%lo0 prefixlen 64 scopeid 0x1
    nd6 options=1<PERFORMNUD>
gif0: flags=8010<POINTOPOINT,MULTICAST> mtu 1280
stf0: flags=0<> mtu 1280
en0: flags=8863<UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST> mtu 1500
    options=2b<RXCSUM,TXCSUM,VLAN_HWTAGGING,TSO4>
    ether 00:1b:63:96:18:38
    inet6 fe80::21b:63ff:fe96:1838%en0 prefixlen 64 scopeid 0x4
    inet 192.168.1.109 netmask 0xffffff00 broadcast 192.168.1.255
    nd6 options=1<PERFORMNUD>
    media: autoselect (1000baseT <full-duplex,flow-control>)
    status: active
en1: flags=8963<UP,BROADCAST,SMART,RUNNING,PROMISC,SIMPLEX,MULTICAST> mtu
1500
    ether 4e:53:50:4f:4f:00
    inet6 fe80::4c53:50ff:fe4f:4f00%en1 prefixlen 64 scopeid 0x5
    inet 169.254.124.206 netmask 0xffff0000 broadcast 169.254.255.255
    nd6 options=1<PERFORMNUD>
    media: autoselect
    status: active
fw0: flags=8863<UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST> mtu 4078
    lladdr 00:1b:63:ff:fe:7e:54:94
    nd6 options=1<PERFORMNUD>
    media: autoselect <full-duplex>
    status: inactive
bridge100: flags=8863<UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST> mtu 1500
    options=3<RXCSUM,TXCSUM>
    ether 02:1b:63:69:b9:64
    inet 192.168.2.1 netmask 0xffffff00 broadcast 192.168.2.255
    inet6 fe80::1b:63ff:fe69:b964%bridge100 prefixlen 64 scopeid 0x7
    Configuration:
        id 0:0:0:0:0:0 priority 0 hellotime 0 fwddelay 0
        maxage 0 holdcnt 0 proto stp maxaddr 100 timeout 1200
        root id 0:0:0:0:0:0 priority 0 ifcost 0 port 0
        ipfilter disabled flags 0x2
    member: en1 flags=3<LEARNING,DISCOVER>
            ifmaxaddr 0 port 5 priority 0 path cost 0
    nd6 options=1<PERFORMNUD>
    media: autoselect
    status: active

Other than Comcast being ****s (which they totally are), I can't see what else could be blocking this traffic.

**Comment by github-hudsucker-com** [comment]
Next step in my troubleshooting is connecting my MBP directly to my bridged Comcase modem. I don't generally like doing that, but hey, troubleshooting.

**Comment by ixalsera** [comment]
Your bridge is definitely up and running and if you're browsing the web fine on other devices, including the n3DS, I can't imagine that your Mac or XZone are the problem. About the only two variables left unaccounted for are port forwarding (is UPnP enabled on your router?) and, as you said, <s>Con</s> Comcast.

I'm looking to push a new build through to the beta track in the next few days that allows you to select the interface you want to share your internet connection from. Once I get that out, you can try tethering your iPhone (it'll have to be via Bluetooth, it seems Apple borked USB tethering in Yosemite, unless I can sort out a workaround) and see if you get any StreetPasses.

**Comment by github-hudsucker-com** [comment]
So minor progress, but at least something.

I set a static IP on my eth connection and then added that IP to the DMZ on my router. I ran the 0.5.4 XZone beta script using Y for worldwide, a five minute rotation, 40 as the starting octet, and NZ@McD01 as the SSID. I can now connect to the German Nintendo Zone, but it's still not collecting any StreetPasses. It's been running for about an hour now and still nothing. I'm going to let it run overnight.

Next up on testing is bypass my router and direct connect to the Comcast modem. Following that, I'm going to take my laptop to a friend's house in a few weeks and try his Internet connection (which is not through Comcast).

**Comment by ixalsera** [comment]
Have you had any success with getting StreetPasses?

**Comment by github-hudsucker-com** [comment]
Sorry for the delayed response. Planning a wedding takes a wee bit of my free time.

I finally got around to hooking up my MBP direct to the Comcast bridged modem and grabbed an address via DHCP, then shared it out via wifi and ran the new 0.6 XZone script. Almost immediately, it loaded up with streetpasses from around the world.

So then I went back to my old set up of modem -> router -> MBP and I'm getting nothing, so there's gotta be something in the router that's blocking this. I'm using a Netgear WNDR3700 v4 running the latest DD-WRT firmware. I've tried enabling/disabling tons of options in the settings and none seem to make any impact.

Do you know if there's any specific LAN settings that would need to be enabled/disabled for this  to work?

**Comment by github-hudsucker-com** [comment]
Also, I may switch to modem -> switch -> router/MBP and see if Comcast allows me to pull multiple IPs (probably not (bastards)) since I'm pretty sure that setup will work.

**Comment by github-hudsucker-com** [comment]
As an update, I switched from the DD-WRT firmware back to the stock Netgear firmware. It's been running for about an hour now and I'm still getting nothing, so I'm wondering if there's something hardware-based in the Netgear WNDR3700v4 router that's blocking this.

**Comment by ixalsera** [comment]
I doubt it would be a hardware issue. Do you have NAT enabled on the router? I'm wondering if a double NAT setup is what's causing havoc. Your Comcast modem will have NAT enabled so all your devices share one external IP with Comcast. If you have NAT enabled on your router too, then all your devices connecting to that router will share the same IP when connecting to the modem (and the modem will assign IPs using DHCP). It could be that this is allowing packets to get lost along the route from Nintendo back to your n3DS.

Just to test this out, see if you can disable NAT on your router (it might be labelled something like "Bridge Mode") as well as DHCP on the router (the modem should be able to handle the DHCP side of things for all your devices) and try to connect again. You should receive StreetPasses in a matter of minutes if it's working. If it doesn't, try setting a static IP on your router too and give it another go.

**Comment by github-hudsucker-com** [comment]
The modem I have is in bridged mode, so the only thing that's doing NAT is my own router. I even re-flashed the firmware in the router back to the stock Netgear firmware last night to rule out the DD-WRT firmware causing problems. Still nothing. The only time this has worked was when the MBP was direct-connected to the Comcast modem and pulled a public IP.

**Comment by ixalsera** [comment]
And if you disable NAT on the router so there is no NAT going on at all (apart from the NAT on your MBP when you enable IS)? It's not an ideal situation, I know, but it helps us rule out possibilities. If that works, perhaps there is a setting on the router that will allow your MBP to bypass NAT while everything else tunnels through it.
